corehq/apps/userreports/tests/test_data_source_citus.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from datetime import datetime
+from unittest import SkipTest
+
+from django.test import TestCase
+
+from corehq.apps.userreports.models import CitusConfig
+from corehq.apps.userreports.tests.utils import (
+    doc_to_change,
+    get_sample_data_source,
+    get_sample_doc_and_indicators,
+)
+from corehq.apps.userreports.util import get_indicator_adapter
+from corehq.pillows.case import get_case_pillow
+
+
+class DataSourceConfigurationCitusDBTest(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(DataSourceConfigurationCitusDBTest, cls).setUpClass()
+        cls.data_source = get_sample_data_source()
+        cls.data_source.engine_id = 'icds-ucr'
+        for indicator in cls.data_source.configured_indicators:
+            if indicator['column_id'] == 'owner':
+                indicator['is_primary_key'] = True
+        cls.data_source.sql_settings.citus_config = CitusConfig(
+            distribution_type="hash",
+            distribution_column="owner"
+        )
+        cls.data_source.sql_settings.primary_key = ['owner', 'doc_id']
+        cls.data_source.save()
+        cls.adapter = get_indicator_adapter(cls.data_source)
+        if not cls.adapter.session_helper.is_citus_db:
+            raise SkipTest("Test only applicable when using CitusDB: {}".format(cls.adapter.session_helper.engine))
+        cls.adapter.build_table()
+
+    def tearDown(self):
+        self.adapter.session_helper.Session.remove()
+        self.adapter.drop_table()
+        self.data_source.delete()
```

Comment:
unrelated to PR: would it be good to make `data_source.delete` automatically call `adapter.drop_table`?

Reply:
that does seem like a reasonable thing to do